### PR TITLE
Merge changes for the builder property to release-1.4 from master.

### PR DESCRIPTION
**Pull request by paulvanbrenk**
This merges the original changes to the completion list from master to release 1.4.

### CODE REVIEW

**Comment by JsonFreeman** [comment]
:+1: Once you merge relevant tests

File: src/services/services.ts [hunk 1]
```diff
@@ -2733,28 +2798,36 @@ module ts {
...
-                            return containingNodeKind === SyntaxKind.Property &&
-                                previousToken.parent.parent.kind === SyntaxKind.InterfaceDeclaration;    // interface a { f; |
+                        case SyntaxKind.DotDotDotToken:
+                            return containingNodeKind === SyntaxKind.Parameter ||
+                                containingNodeKind === SyntaxKind.Constructor;
 
                         case SyntaxKind.PublicKeyword:
                         case SyntaxKind.PrivateKeyword:
-                        case SyntaxKind.StaticKeyword:
-                        case SyntaxKind.DotDotDotToken:
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: Why did you remove these?

Comment (DanielRosenwasser) [inline src/services/services.ts]: `...` might still need to be here.
